packages/stateless/src/lib/provider/provider.interface.ts

```diff
@@ -56,11 +56,40 @@ export interface ISendMessageSuccessResponse {
   date?: string;
 }
 
+export enum EmailEventStatusEnum {
+  OPEN = 'open',
+  DELIVERY = 'delivery',
+  BOUNCE = 'bounce',
+  DROPPED = 'dropped',
+  CLICK = 'click',
+}
+
+export interface IEventBody {
+  status: string;
+  date: string;
+  externalId?: string;
+  attempts?: number;
+  response?: string;
+  // Contains the raw content from the provider webhook
+  row?: string;
+}
+
+export interface IEmailEventBody extends IEventBody {
+  status: EmailEventStatusEnum;
+}
+
 export interface IEmailProvider extends IProvider {
   channelType: ChannelTypeEnum.EMAIL;
 
   sendMessage(options: IEmailOptions): Promise<ISendMessageSuccessResponse>;
 
+  getMessageId?: (body: any) => string | string[];
```

Comment:
I would totally push to force this method to only return an array of strings rather than allowing the 2 different types, as it would complicate the processing for consumers of this method. So for the flows where only one message is returned to make it as single element of an array.
```suggestion
  getMessageId?: (body: any) => string[];
```